google/cloud/pubsub/publisher_connection_test.cc

```diff
@@ -342,34 +347,71 @@ TEST(PublisherConnectionTest, HandleTransientEnabledRetry) {
 using ::google::cloud::testing_util::DisableTracing;
 using ::google::cloud::testing_util::EnableTracing;
 using ::google::cloud::testing_util::InstallSpanCatcher;
+using ::google::cloud::testing_util::OTelContextCaptured;
 using ::google::cloud::testing_util::SpanNamed;
+using ::google::cloud::testing_util::ThereIsAnActiveSpan;
 using ::testing::IsEmpty;
 using ::testing::UnorderedElementsAre;
 
 TEST(MakePublisherConnectionTest, TracingEnabled) {
   auto span_catcher = InstallSpanCatcher();
   auto mock = std::make_shared<pubsub_testing::MockPublisherStub>();
   Topic const topic("test-project", "test-topic");
+
+  auto mock_message_batch =
+      std::make_shared<pubsub_testing::MockMessageBatch>();
+  std::mutex mu;
+  std::condition_variable cv;
+  bool is_publish_complete = false;  // ABSL_GUARDED_BY(mu)
```

Comment:
nit: you can simplify this even further:

```suggestion
  std::promise<void>  publish_complete;
```

Reply:
You can use `google::cloud::promise<void>` if you prefer.

Reply:
Done